src/SelfUpdateCommand.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Robo;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem as sfFilesystem;
+
+/**
+ * Update the robo.phar from the latest github release
+ *
+ * @author Alexander Menk <alex.menk@gmail.com>
+ */
+class SelfUpdateCommand extends Command
+{
+    private $command;
+
+    protected $gitHubRepository;
+
+    protected $currentVersion;
+
+    public function __construct($name = null, $currentVersion = null, $gitHubRepository = null)
+    {
+        parent::__construct($name);
+        $this->currentVersion = $currentVersion;
+        $this->gitHubRepository = $gitHubRepository;
+    }
+
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setAliases(array( 'selfupdate' ))
+            ->setDescription('Updates the robo.phar to the latest version.')
+            ->setHelp(
+                <<<EOT
+The <info>self-update</info> command checks github for newer
+versions of robo and if found, installs the latest.
+EOT
+            );
+    }
+
+    protected function getLatestReleaseFromGithub()
+    {
+        $opts = [
+            'http' => [
+                'method' => 'GET',
+                'header' => [
+                    'User-Agent: ' . Robo::APPLICATION_NAME . ' Self-Update (PHP)'
+                ]
+            ]
+        ];
+
+        $context = stream_context_create($opts);
+
+        $releases = file_get_contents('https://api.github.com/repos/' . $this->gitHubRepository . '/releases', false, $context);
+        $releases = json_decode($releases);
+
+        if (! isset($releases[0])) {
+            throw new \Exception('API error - no release found at GitHub repository ' . $this->gitHubRepository);
+        }
+
+        $version = $releases[0]->tag_name;
+        $url     = $releases[0]->assets[0]->browser_download_url;
+
+        return [ $version, $url ];
+    }
+
+    public function isEnabled()
+    {
+        return !empty(\Phar::running());
```

Comment:
I don't have time to test this right now, but I presume you did? At one point in the past, `Phar::running()` was returning false regardless of whether Robo was running as a Phar or not.

Reply:
I tested it